packages/block-library/src/post-title/edit.js

```diff
@@ -114,20 +119,22 @@ export default function PostTitleEdit( {
 
 	return (
 		<>
-			<BlockControls group="block">
-				<HeadingLevelDropdown
-					selectedLevel={ level }
-					onChange={ ( newLevel ) =>
-						setAttributes( { level: newLevel } )
-					}
-				/>
-				<AlignmentControl
-					value={ textAlign }
-					onChange={ ( nextAlign ) => {
-						setAttributes( { textAlign: nextAlign } );
-					} }
-				/>
-			</BlockControls>
+			{ blockEditingMode === 'default' && (
+				<BlockControls group="block">
```

Comment:
<img width="228" alt="Screenshot 2023-05-31 at 3 32 14 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/a840c4d1-5e17-41c1-8ec7-07a6167c9cab">

The black dot is a collapsed border style, and appears on all page blocks that are `contentOnly`, e.g., feature image block as well.

There's a permanent border on the following element (I've added `border:0` inline to remove it via the console)

```html
<div id="id-0f1fdc" role="toolbar" aria-orientation="horizontal" aria-label="Block tools" class="components-accessible-toolbar block-editor-block-contextual-toolbar" style="
    border: 0;
"><div class="block-editor-block-toolbar"><div></div></div></div>
```

This element is rendered by `<BlockContextualToolbar />` ([block-contextual-toolbar.js](https://github.com/WordPress/gutenberg/blob/9eba2068716dce9f79c1190c60ac6ff1f2de6689/packages/block-editor/src/components/block-tools/block-contextual-toolbar.js#L82-L82)). The offending CSS rule for `.block-editor-block-contextual-toolbar` is in [style.scss](https://github.com/WordPress/gutenberg/blob/9eba2068716dce9f79c1190c60ac6ff1f2de6689/packages/block-editor/src/components/block-tools/style.scss#L98-L98).

The element is always rendered, even when the slots are empty.  

I'm not sure what the best approach is. Checking via `useSlotFills()` whether the slot is, indeed filled? Would we have to loop through all [groups.js](https://github.com/WordPress/gutenberg/blob/9eba2068716dce9f79c1190c60ac6ff1f2de6689/packages/block-editor/src/components/block-controls/groups.js#L12-L12)?

I'm also guessing the value of `blockEditingMode` in  <BlockContextualToolbar />` might not be useful to since the toolbar can show for `default` and `contentOnly`. In this case the site title block has a value of `contentOnly`, but the slots are not filled.

Just dumping my brain here in case it helps... :)

Reply:
Could we do something like :has( > div:empty:only-child)? Fine to address as a follow-up though 😅

Reply:
`:has` doesn't work in Firefox 😭 I'll pick your brain about this next week.